### PR TITLE
I1473 lock conf

### LINKED ISSUE #1473: Lock files should be in a separate dir and turn on/off through config
Create $dataDir/lock dir, and put all lock files there. Also introduce `smv.lockDir` and `smv.lock` options.

### PR DESCRIPTION
**Pull request by ninjapapa**
Fixed #1473

### CODE REVIEW

File: src/main/python/smv/smvlock.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            self._lock_path,
+            3600 * 1000 
+        )
+        self.slock.lock()
+        return None
+
+    def __exit__(self, type, value, traceback):
+        self.slock.unlock()
+
+class NonOpLock(object):
```
Comment (AliTajeldin) [inline src/main/python/smv/smvlock.py]: neato

File: src/main/python/smv/smviostrategy.py [hunk 2]
```diff
@@ -92,19 +83,8 @@ def _write(self, raw_data):
...
-            else:
-                self.smvApp.log.info("No cached result found. Caching result at {}".format(self._file_path))
-                # Delete outputs in case data was partially written previously
-                # since `isPersisted` test on schema file, this case only happens when schema was written half way
-                self.remove()
-                self._write(dataframe)
-        finally:
-            slock.unlock()
+        # May add lock or other logic here in future
+        self._write(dataframe)
```
Comment (AliTajeldin) [inline src/main/python/smv/smviostrategy.py]: where do we acquire the lock for the data write?

Comment (ninjapapa) [reply src/main/python/smv/smviostrategy.py]: Current file based lock is acquired from the caller. Maybe in the future the write itself has some other way to lock, maybe not. Should we just remove `_write` and `_read` and use the one without underscore?

Comment (AliTajeldin) [reply src/main/python/smv/smviostrategy.py]: Yah, I like that better as `_xxx` implies that `xxx` is somewhat private (by convention)

Comment (AliTajeldin) [reply src/main/python/smv/smviostrategy.py]: Let's discuss location of lock/unlock tomorrow.  Would be in another ticket if we need to change anyway.  Merging this.

**Pull request merged**